Src/VimCore/CoreInterfaces.fs

```diff
@@ -1509,61 +1509,82 @@ type KeyMappingResult =
 [<DebuggerDisplay("{ToString()}")>]
 type CharacterSpan = 
 
-    val private _start: SnapshotPoint
+    val private _start: VirtualSnapshotPoint
 
     val private _lineCount: int
 
     val private _lastLineLength: int
 
+    val private _useVirtualSpace: bool

```

Comment:
Having trouble understanding of this means

1. Consumer should consider the virtual space when consuming this type 
2. This type was created by considering virtual space of the inputs

Reply:
I will add a explanatory comment.

It means "This character span was created when virtual space was in effect and should/will use the virtual interpretation of start point and length."

If you have a character span that selects four characters starting at the `c` in a line consisting only of `cat`, then the meaning of the character span depends on whether you are using virtual space. If you are using virtual space, the span ends in a space. Otherwise the span ends in a newline.

In other words, when using virtual space, an end point in virtual space always adds spaces whereas when not using virtual space, an end point in virtual space (which can only be just past the end of the line) includes the linebreak of that line.

Last way to explain it, although the selection looks identical in vim, the `virtualedit` setting can affect what gets yanked.